src/main/java/net/minecraftforge/datafix/ForgeDataFixerEventHandler.java

```diff
@@ -0,0 +1,169 @@
+package net.minecraftforge.datafix;
+
+import com.google.common.collect.Lists;
+import com.mojang.datafixers.schemas.Schema;
+import net.minecraft.Util;
+import net.minecraft.util.datafix.DataFixers;
+import net.minecraftforge.event.datafix.ConfigureDatafixSchemaEvent;
+import net.minecraftforge.fml.IModStateTransition;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.stream.Stream;
+
+/**
+ * Event handler utility class for managing the events related to data fixers.
+ * This particular class is used to handle firing of the parallel mod loading events
+ * for the DFU setup.
+ *
+ * These events are fired immediately after mod construction and before the rest of the game
+ * loads, even before registries are fired.
+ */
+public class ForgeDataFixerEventHandler
+{
+    private static final Logger LOGGER = LogManager.getLogger();
+
+    /**
+     * Gives access to the list of schemas registered with the data fixers.
+     * If we are currently running an incompatible version of DFU or somebody messed with it,
+     * then an empty list is returned.
+     *
+     * @return The list of all known schemas to handle.
+     */
+    private static List<ForgeSchema> getAllSchemas()
+    {
+        //Check if we are running in a compatible environment.
+        if (DataFixers.getDataFixer() instanceof ForgeDataFixerDelegateHandler forgeDataFixerDelegateHandler)
+        {
+            //Forge DFU wrapper is active, grabbing its configured schemas.
+            return forgeDataFixerDelegateHandler.getAllSchemas();
+        }
+
+        //Not compatible environment, return an empty list.
+        return Collections.emptyList();
+    }
+
+    /**
+     * Builder function for creating the events that are fired on start up.
+     * This returns one generator per schema, and keeps the schema in the order defined.
+     *
+     * The generator will always return the same multi-thread capable event reference regardless
+     * of the passed in mod name.
+     *
+     * @return The stream of event generators used to fire one event per mod per schema.
+     */
+    public static Stream<IModStateTransition.EventGenerator<?>> generateConfigureEvents()
+    {
+        //Grab the schemas
+        List<ForgeSchema> schemas = Lists.newArrayList(getAllSchemas());
```

Comment:
Is there a reason this isn't just `new ArrayList<>(getallSchemas())`

Reply:
Not really I just do it that way.......